src/components/CollectionDetails/index.tsx

```diff
@@ -218,6 +260,29 @@ const CollectionDetails = ({ collection }: CollectionDetailsProps) => {
               ) && (
                 <StatusBadge
                   status={collectionStatus4k}
+                  downloadItem={downloadStatus[0]}
+                  downloadBlock={
+                    <ul>
+                      {downloadStatus.map((status, index) => (
+                        <li
+                          key={`dl-status-${status.externalId}-${index}`}
+                          className="border-b border-gray-700 last:border-b-0"
+                        >
+                          <DownloadBlock
+                            downloadItem={status}
+                            title={data.parts
+                              .filter(
+                                (item) =>
+                                  item.mediaInfo?.externalServiceId4k ===
+                                  status.externalId
+                              )
+                              .map((title) => title.title)
+                              .join('')}
+                          />
+                        </li>
+                      ))}
+                    </ul>
```

Comment:
Feels strange we are duplicating this everywhere we use the `StatusBadge`. Can we just move this into `StatusBadge` itself?